docs/release_notes.rst

```diff
@@ -116,11 +116,19 @@ Plant Parts List Module Changes
   :mod:`pudl.analysis.mcoe.DEFAULT_GENS_COLS`. If additional columns that are not part
   of the default list are needed from the EIA 860 generators table, these columns can be
   passed in with the ``gens_cols`` argument.  See :pr:`1550`
+* For greater memory efficiency, appropriate columns are now cast to string and
+  categorical types when the full plant parts list is created.
 
 Metadata
 ^^^^^^^^
 * Used the data source metadata class added in release 0.6.0 to dynamically generate
   the data source documentation (See :doc:`data_sources/index`). :pr:`1532`
+* Column attributes may now be the pandas ``Categorical`` data type. Using categorical
```

Comment:
I think this will break the system as it's implemented right now.

Column types are independent of Pandas (and SQLite, and Parquet, etc) and are defined by the [TableSchema specificaton](https://specs.frictionlessdata.io/table-schema/#field-descriptors). We then define mappings from that generic TableSchema type to the specific types for the various outputs (Pandas, SQLite, pyarrow, etc.).

It seems like the desired functionality is an unenumerated categorical column? Where we're telling the system "Please treat this as a categorical value, but figure out what the categories are dynamically based on the contents of the column, without imposing any constraints on the values."

This seems like it should not be implemented in the column type (since there is no categorical type under the standard, just enum constrained values of a given type), but rather in the process of translating to the various output formats.  Somehow we need to communicate to the output that some columns should be treated as categoricals when they're put into a dataframe or parquet file, even if they aren't constrained.